core/common/src/main/java/alluxio/retry/TimeBoundedRetry.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.retry;
+
+import alluxio.time.Sleeper;
+import alluxio.time.TimeContext;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+
+/**
+ * Retry mechanism which performs retries until a certain period of time has elapsed. Subclasses
+ * determine the interval between retries.
+ */
+public abstract class TimeBoundedRetry implements RetryPolicy {
```

Comment:
`AbstractTimeBoundedRetry` to be consistent with our existing naming convention?

Reply:
I thought the naming convention is to prefer descriptive names where possible, and only use generic terms like `Abstract` or `Default` when we can't come up with an available, concise, and descriptive name. From a practical angle, I think it's easy enough to tell that the class is abstract, and it's not worth making the name more verbose.

Doing a quick check on the codebase shows about 50% following the `AbstractX` pattern for abstract classes:
```
➜  alluxio git:(cc4ac16e32) ✗ git grep 'public abstract class' | grep Abstract | wc -l
      25
➜  alluxio git:(cc4ac16e32) ✗ git grep 'public abstract class' | grep -v Abstract | wc -l
      22
```

Reply:
SG